core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -374,4 +385,107 @@ static TableMetadata commit(TableOperations ops, UpdateTableRequest request) {
 
     return ops.current();
   }
+
+  public static BaseView baseView(View view) {
+    Preconditions.checkArgument(view instanceof BaseView, "View must be a BaseView");
+    return (BaseView) view;
+  }
+
+  public static ListTablesResponse listViews(ViewCatalog catalog, Namespace namespace) {
+    return ListTablesResponse.builder().addAll(catalog.listViews(namespace)).build();
+  }
+
+  public static LoadViewResponse createView(
+      ViewCatalog catalog, Namespace namespace, CreateViewRequest request) {
+    request.validate();
+
+    ViewMetadata viewMetadata = request.metadata();
```

Comment:
This should not pass full view metadata. It should pass the parts of view metadata that are used to create a view, just like tables.

Otherwise, the contract for this route is confusing and loses data. The code below only uses `currentVersion`, but what if `currentVersion` is not the only version?

Reply:
agreed, I was planning to update `CreateViewRequest` to be change-based rather than sending the full metadata. That's also why I marked the PR as WIP